lib/rules/computed-property-return.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @fileoverview Enforces that a return statement is present in computed property (computed-property-return)
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function create (context) {
+  const options = context.options[0] || {}
+  const treatUndefinedAsUnspecified = !(options.treatUndefinedAsUnspecified === false)
+
+  let funcInfo = {
+    funcInfo: null,
+    codePath: null,
+    hasReturn: false,
+    hasReturnValue: false,
+    node: null
+  }
+  const forbiddenNodes = []
+
+  // ----------------------------------------------------------------------
+  // Helpers
+  // ----------------------------------------------------------------------
+  function isValidReturn () {
+    return (!treatUndefinedAsUnspecified && funcInfo.hasReturn) || (treatUndefinedAsUnspecified && funcInfo.hasReturnValue)
+  }
+
+  // ----------------------------------------------------------------------
+  // Public
+  // ----------------------------------------------------------------------
+
+  return Object.assign({},
+    {
+      onCodePathStart (codePath, node) {
+        funcInfo = {
+          codePath,
+          funcInfo: funcInfo,
+          hasReturn: false,
+          hasReturnValue: false,
+          node
+        }
+      },
+      onCodePathEnd () {
+        funcInfo = funcInfo.funcInfo
+      },
+      ReturnStatement (node) {
+        funcInfo.hasReturn = true
+        funcInfo.hasReturnValue = Boolean(node.argument)
+      },
+      'FunctionExpression:exit' (node) {
+        if (!isValidReturn() && funcInfo.codePath.currentSegments.some((segment) => segment.reachable)) {
+          forbiddenNodes.push({
+            hasReturn: funcInfo.hasReturn,
+            node: funcInfo.node,
+            type: 'return'
+          })
+        }
+      }
+    },
+    utils.executeOnVueInstance(context, properties => {
```

Comment:
Shouldn't it be `executeOnVueComponent` ? And this one in `computed-data` rule instead?